core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -903,7 +1023,7 @@ private boolean checkForFullMatch(InFlightBatch inFlightBatch, long firstOffsetT
      * @return True if the start offset has moved and within the request first and last offset, false otherwise.
      */
     private boolean checkForStartOffsetWithinBatch(long batchFirstOffset, long batchLastOffset) {
-        return batchFirstOffset < startOffset && batchLastOffset >= startOffset;
+        return batchFirstOffset < startOffset() && batchLastOffset >= startOffset();
```

Comment:
👍 though the method is being called from methods which already has lock but to make it safe you might want to use a single variable which first reads `startOffset()` and then is used. So we make sure that value is always same and not changed within 2 different calls to `startOffset()`.

Reply:
I won't suggest that, the method should be defined in a way that it's thread safe. I ll suggest somehting like below:

```
long localStartOffset = startOffset();
return batchFirstOffset < localStartOffset && batchLastOffset >= localStartOffset;
```